tesseract/tesseract_planning/tesseract_motion_planners/include/tesseract_motion_planners/trajopt/config/trajopt_planner_default_config.h

```diff
@@ -90,12 +90,16 @@ struct TrajOptPlannerDefaultConfig : public TrajOptPlannerConfig
 
   /** @brief The type of contact test to perform: FIRST, CLOSEST, ALL */
   tesseract_collision::ContactTestType contact_test_type = tesseract_collision::ContactTestType::ALL;
-  /** @brief If true, collision checking will be enabled. Default: true*/
+  /** @brief If true, collision checking will be enabled and a collision constraint term will be added to the problem. Default: true*/
   bool collision_check = true;
+  /** @brief If true, a collision cost term will be added to the problem. Default: false*/
+  bool use_collision_cost_term = false;
   /** @brief If true, use continuous collision checking */
   trajopt::CollisionEvaluatorType collision_type = trajopt::CollisionEvaluatorType::CAST_CONTINUOUS;
-  /** @brief Max distance over which collisions are checked */
+  /** @brief Max distance in which collisions will be evaluated as constraints */
   double collision_safety_margin = 0.025;
+  /** @brief Max distance in which collisions will be evaluated as costs if use_collision_cost_term is true */
+  double collision_buffer_margin = 0.05;
```

Comment:
I recommend have different parameters for each.

Constraint Config. If you want to put these in a struct that is also ok.
```
collision_constraint_check
collision_constraint_type
collision_constraint_safety_margin
collision_constraint_safety_margin_buffer
collision_constraint_coeff
```
or 
```
struct CollisionConstraintConfig
{
  bool check = true;
  trajopt::CollisionEvaluatorType type = trajopt::CollisionEvaluatorType::CAST_CONTINUOUS;
  double safety_margin = 0.01;
  double safety_margin_buffer = 0.05;
  double coeff = 20;
}
```

Cost Config. If you want to put these in a struct that is also ok.
```
collision_cost_check
collision_cost_type
collision_cost_buffer_margin
collision_cost_coeff
```
or 
```
struct CollisionCostConfig
{
  bool check = true;
  trajopt::CollisionEvaluatorType type = trajopt::CollisionEvaluatorType::CAST_CONTINUOUS;
  double buffer_margin = 0.025;
  double coeff = 20;
}
```

Reply:
Maybe instead of check we change to enabled?

Reply:
I added some config structs to separately wrap settings for costs and constraints.